index.bs

```diff
@@ -78,4 +251,313 @@ interface StorageBucket {
 };
 </xmp>
 
+A {{StorageBucket}} has an associated [=/storage bucket=].
+
+<h3 id="storage-bucket-persistence">Persistence</h3>
+
+Issue: Merge with [[Storage#buckets]] which already defines [=bucket mode=].
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">persist()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with {{InvalidStateError}}.
+
+    1. Otherwise,
+
+        1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+        1. Let |permission| be the result of [=getting the current permission state=] with `"persistent-storage"` and |environment|.
```

Comment:
Can we short-circuit if the bucket already has persistent set to true?  Like, what happens if persist() is called twice.